From 4236eff110386b01eee254effe3f287023303a33 Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Thu, 30 Apr 2015 13:29:54 +0200
Subject: [PATCH 090/117] inspection: read UUID and label for xfs
 (RHBZ#1216305)

Read UUID and label also for xfs filesystems in the old inspection
code.
get_e2uuid and get_e2label are actually deprecated names for vfs_uuid
and vfs_label, which use blkid, so working for xfs as well.
---
 perl/lib/Sys/Guestfs/Lib.pm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/perl/lib/Sys/Guestfs/Lib.pm b/perl/lib/Sys/Guestfs/Lib.pm
index ce46b4f..e9a91e3 100644
--- a/perl/lib/Sys/Guestfs/Lib.pm
+++ b/perl/lib/Sys/Guestfs/Lib.pm
@@ -443,8 +443,8 @@ sub inspect_partition
         $r{is_swap} = 1;
     }
 
-    # If it's ext2/3/4, then we want the UUID and label.
-    if (exists $r{fstype} && $r{fstype} =~ /^ext/) {
+    # If it's ext2/3/4 or xfs, then we want the UUID and label.
+    if (exists $r{fstype} && ($r{fstype} =~ /^ext/ || $r{fstype} eq 'xfs')) {
         $r{uuid} = $g->get_e2uuid ($dev);
         $r{label} = $g->get_e2label ($dev);
     }
-- 
2.5.0

